tests/test_datasets/test_dataset.py

```diff
@@ -171,14 +83,80 @@ def test_get_data_pandas(self):
     def test_get_data_boolean_pandas(self):
         # test to check that we are converting properly True and False even
         # with some inconsistency when dumping the data on openml
-        data = self.jm1.get_data(dataset_format='dataframe')
+        data, _, _, _ = self.jm1.get_data()
         self.assertTrue(data['defects'].dtype.name == 'category')
-        self.assertTrue(
-            set(data['defects'].cat.categories) == set([True, False])
-        )
-        data = self.pc4.get_data(dataset_format='dataframe')
+        self.assertTrue(set(data['defects'].cat.categories) == {True, False})
+
+        data, _, _, _ = self.pc4.get_data()
         self.assertTrue(data['c'].dtype.name == 'category')
-        self.assertTrue(set(data['c'].cat.categories) == set([True, False]))
+        self.assertTrue(set(data['c'].cat.categories) == {True, False})
+
+    def test_get_data_no_str_data_for_nparrays(self):
+        # check that an error is raised when the dataset contains string
+        err_msg = "PyOpenML cannot handle string when returning numpy arrays"
+        with pytest.raises(PyOpenMLError, match=err_msg):
+            self.titanic.get_data(dataset_format='array')
+
+    def test_get_data_with_rowid(self):
+        self.dataset.row_id_attribute = "condition"
+        rval, _, categorical, _ = self.dataset.get_data(include_row_id=True)
+        self.assertEqual(rval.shape, (898, 39))
+        self.assertEqual(len(categorical), 39)
+
+        rval, _, categorical, _ = self.dataset.get_data()
+        self.assertEqual(rval.shape, (898, 38))
+        self.assertEqual(len(categorical), 38)
```

Comment:
Hm, what do you think about adding some type checks here?

Reply:
I am a little annoyed by the fact that this file seems to be 80% type checking in almost the same format every time.
But you are right that type checking is probably sensible here, in particular the column types as if anything went wrong with omitting the column, column types might be mismatched too.

I guess this is something to refactor at some point though.